File: app/python/plugins/ai_serviter.py
from lattice import Capability, ModuleSpec
from ai_serviter import AIServiter


def start_ai_serviter(ctx):
    root = ctx.config.get("project_root", ".")
    ctx.bind(Capability("ai_serviter", (1, 0)), AIServiter(root, config=ctx.config))


REGISTER_MODULES = [
    ModuleSpec(
        name="ai-serviter",
        provides=[Capability("ai_serviter", (1, 0))],
        requires=[],
        start=start_ai_serviter,
    )
]

REGISTER_ADAPTERS = []

File: typescript/src/plugins/aiServiter.ts
import type { ModuleSpec } from "../core.js";
import { AIServiter } from "../serviter/serviter.js";

export const REGISTER_MODULES: ModuleSpec[] = [
  {
    name: "ai-serviter",
    provides: [{ name: "ai_serviter", version: [1, 0] }],
    requires: [],
    start(ctx) {
      const root = (ctx.config.projectRoot as string | undefined) ?? ".";
      ctx.bind({ name: "ai_serviter", version: [1, 0] }, new AIServiter(root));
    },
  },
];

export const REGISTER_ADAPTERS = [];
